Modules/Loadable/Segmentations/Widgets/qMRMLSegmentEditorWidget.cxx

```diff
@@ -3366,3 +3424,103 @@ void qMRMLSegmentEditorWidget::setAutoShowMasterVolumeNode(bool autoShow)
     }
   d->AutoShowMasterVolumeNode = autoShow;
 }
+
+//---------------------------------------------------------------------------
+void qMRMLSegmentEditorWidget::updateSliceRotateWarningButtonVisibility()
+{
+  Q_D(qMRMLSegmentEditorWidget);
+  vtkNew<vtkMatrix4x4> segmentationIJKToRAS;
+  if (!d->segmentationIJKToRAS(segmentationIJKToRAS))
+    {
+    // segmentation orientation cannot be determined
+    d->SliceRotateWarningButton->hide();
+    return;
+    }
+
+  qSlicerLayoutManager* layoutManager = qSlicerApplication::application()->layoutManager();
+  if (!layoutManager)
+    {
+    // application is closing
+    d->SliceRotateWarningButton->hide();
+    return;
+    }
+
+  // Check if any of the slices are rotated
+  bool sliceRotated = false;
+  foreach(QString sliceViewName, layoutManager->sliceViewNames())
+    {
+    qMRMLSliceWidget* sliceWidget = layoutManager->sliceWidget(sliceViewName);
+    if (!d->segmentationDisplayableInView(sliceWidget->mrmlSliceNode()))
+      {
+      continue;
+      }
+    vtkMRMLSliceNode* sliceNode = sliceWidget->mrmlSliceNode();
+    if (!sliceNode)
+      {
+      continue;
+      }
+    vtkMatrix4x4* sliceToRAS = sliceNode->GetSliceToRAS();
+    // Only need to check alignment of X and Y axes, if they are aligned then Z axis will be aligned, too
+    for (int sliceAxisIndex = 0; sliceAxisIndex < 2; ++sliceAxisIndex)
+      {
+      double sliceAxisDirection[3];
```

Comment:
We agreed it's good practice to initialize vectors, it will be nice to show good example.